gui/default/untrusted/syncthing/core/syncthingController.js

```diff
@@ -1911,36 +1912,48 @@ angular.module('syncthing.core')
             }
         };
 
+        function addFolderInit(recvEnc) {
+            $scope.editingExisting = false;
+            $scope.currentFolder = angular.copy($scope.folderDefaults);
+            if (recvEnc) {
+                $scope.currentFolder.type = "receiveencrypted";
+                $scope.currentFolder.ignorePerms = true;
+                delete $scope.currentFolder.versioning;
+            }
+            initShareEditing('folder');
+            $scope.ignores.text = '';
+            $scope.ignores.error = null;
+            $scope.ignores.disabled = false;
+        }
+
         $scope.addFolder = function () {
             $http.get(urlbase + '/svc/random/string?length=10').success(function (data) {
-                $scope.editingExisting = false;
-                $scope.currentFolder = angular.copy($scope.folderDefaults);
-                initShareEditing('folder');
+                addFolderInit(false);
                 $scope.currentFolder.id = (data.random.substr(0, 5) + '-' + data.random.substr(5, 5)).toLowerCase();
                 $scope.currentSharing.unrelated = $scope.otherDevices();
-                $scope.ignores.text = '';
-                $scope.ignores.error = null;
-                $scope.ignores.disabled = false;
                 $scope.editFolderModal();
             });
         };
 
         $scope.addFolderAndShare = function (folder, folderLabel, device) {
-            $scope.editingExisting = false;
-            $scope.currentFolder = angular.copy($scope.folderDefaults);
+            var pendingFolder = $scope.pendingFolders[folder];
```

Comment:
If the addFolderAndShare function needs access to the pendingFolder's details, we could pass the corresponding `offeredBy[k]` object instead of the folderLabel parameter.